### PR TITLE
A dummy PR to practice PR reviews

### PR DESCRIPTION
**Pull request by damithc**
This is a dummy PR to practice PR reviews.
The code added has multiple coding standard violations, which are expected to be detected by reviewers.

### CODE REVIEW

File: src/main/java/Task.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    private boolean important;
+    List<String> pastDescription = new ArrayList<>(); // a list of past descriptions
+
+    public Task(String d) {
+      this.description = d;
+      if (!d.isEmpty())
+          this.important = true;
+    }
+
+    public String getAsXML() { return "<task>"+description+"</task>"; }
```
Comment (skyventus) [inline src/main/java/Task.java:15]: Is this consider magic string? Should you assign to a string and return?

File: src/main/java/Task.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import java.util.*;
+
+public class Task {
+    public static final String descriptionPrefix = "description: ";
```
Comment (adi-kd0021) [inline src/main/java/Task.java:4]: Constants variable name should be all Uppercase ?

File: src/main/java/Duke.java [hunk 3]
```diff
@@ -6,5 +6,7 @@ public static void main(String[] args) {
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        Task my_task = new Task("test task");
```
Comment (damithc) [inline src/main/java/Duke.java:9]: Should this be `myTask`? :thinking:

File: src/main/java/Task.java (hunk 2 above)
Comment (damithc) [inline src/main/java/Task.java:4]: Should the variable name be in SCREAMING_SNAKE case, because it's a constant?

File: src/main/java/Task.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import java.util.*;
```
Comment (damithc) [inline src/main/java/Task.java:1]: Coding standard says never to use wildcard imports.

**Review by skyventus — COMMENTED** [review]
Comment on code violation

**Review by damithc — COMMENTED** [review]
The code is generally OK. But I spotted some coding standard violations. There might be more -- I didn't check the whole code. Perhaps you can find and fix all of them and update the PR?

File: src/main/java/Task.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+public class Task {
+    public static final String descriptionPrefix = "description: ";
+    private String description;
+    private boolean important;
```
Comment (wz-chan) [inline src/main/java/Task.java:6]: Should this be isImportant?

File: src/main/java/Task.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+      if (!d.isEmpty())
+          this.important = true;
+    }
+
+    public String getAsXML() { return "<task>"+description+"</task>"; }
+
+    /**
+     * Print the description as a string.
+     */
+    public void printingDescription(){ System.out.println(this); }
```
Comment (linqing42) [inline src/main/java/Task.java:20]: Is it better to avoid the unnecessary use of this?

Comment (wz-chan) [inline src/main/java/Task.java:20]: Should this be printDescription()?

File: src/main/java/Task.java (hunk 1 above)
Comment (wz-chan) [inline src/main/java/Task.java:15]: Should this be return "\<task>" + description + "\</task>";

File: src/main/java/Task.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import java.util.*;
+
+public class Task {
+    public static final String descriptionPrefix = "description: ";
+    private String description;
+    private boolean important;
+    List<String> pastDescription = new ArrayList<>(); // a list of past descriptions
+
+    public Task(String d) {
+      this.description = d;
```
Comment (wz-chan) [inline src/main/java/Task.java:10]: Is the this keyword redundant?

File: src/main/java/Task.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+      if (!d.isEmpty())
+          this.important = true;
```
Comment (wz-chan) [inline src/main/java/Task.java:12]: Is the this keyword redundant?

**Review by wz-chan — COMMENTED** [review]
Hi! Overall, I think the changes are good. I have made some recommendations so that the code fulfills Java Coding Standard.

File: src/main/java/Task.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
+import java.util.*;
+
+public class Task {
+    public static final String descriptionPrefix = "description: ";
+    private String description;
+    private boolean important;
+    List<String> pastDescription = new ArrayList<>(); // a list of past descriptions
+
+    public Task(String d) {
```
Comment (dgc5213) [inline src/main/java/Task.java:9]: Should the variable (String d) name be in meaningful with specific context?

File: src/main/java/Task.java (hunk 5 above)
Comment (dgc5213) [inline src/main/java/Task.java:6]: should it explain the named entity to the reader accurately and at a sufficient level of detail? boolean important, should it changed to 'isImportant' ?

File: src/main/java/Duke.java (hunk 3 above)
Comment (HongJiancheng) [reply src/main/java/Duke.java:9]: I feel it should be `myTask`, too.

File: src/main/java/Task.java (hunk 5 above)
Comment (adi-kd0021) [inline src/main/java/Task.java:6]: Boolean variables/methods should be named to sound like booleans ? 

Reference: https://se-education.org/guides/conventions/java/index.html

File: src/main/java/Task.java [hunk 10]
```diff
@@ -0,0 +1,24 @@
+import java.util.*;
+
+public class Task {
+    public static final String descriptionPrefix = "description: ";
+    private String description;
+    private boolean important;
+    List<String> pastDescription = new ArrayList<>(); // a list of past descriptions
```
Comment (adi-kd0021) [inline src/main/java/Task.java:7]: Plural form should be used on names representing a collection of objects ?

File: src/main/java/Task.java (hunk 1 above)
Comment (adi-kd0021) [inline src/main/java/Task.java:15]: Abbreviations and acronyms should not be uppercase when used as a part of a name.

public String getAsXml() would be more appropriate here, please check this ?

File: src/main/java/Task.java (hunk 6 above)
Comment (adi-kd0021) [inline src/main/java/Task.java:20]: Usage of the **this** keyword only when a field is shadowed by a method or constructor parameter. Here the **this** a little ambiguous.

**Review by adi-kd0021 — COMMENTED** [review]
Practice review was done with reference to the coding conventions/ standards below. 

https://se-education.org/guides/conventions/java/index.html

File: src/main/java/Task.java (hunk 9 above)
Comment (tototto) [inline src/main/java/Task.java:9]: poor choice of variable name for task

File: src/main/java/Task.java (hunk 1 above)
Comment (tototto) [inline src/main/java/Task.java:15]: return string should be stored in a meaningful variable name. That variable should then be used.

File: src/main/java/Task.java (hunk 6 above)
Comment (tototto) [inline src/main/java/Task.java:20]: This line can be refactored out

File: src/main/java/Duke.java (hunk 3 above)
Comment (Impala36) [inline src/main/java/Duke.java:9]: Should my_task be in camelCase?

File: src/main/java/Task.java (hunk 5 above)
Comment (Impala36) [inline src/main/java/Task.java:6]: Should a verb accompany the boolean?

File: src/main/java/Task.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+
+    public Task(String d) {
+      this.description = d;
+      if (!d.isEmpty())
```
Comment (Impala36) [inline src/main/java/Task.java:11]: Perhaps some Egyptian brackets with the if statement here?

File: src/main/java/Task.java (hunk 1 above)
Comment (Impala36) [inline src/main/java/Task.java:15]: Maybe having the return statement and the closing bracket on a new line would be a good idea? As well as the toString() function below.

File: src/main/java/Task.java (hunk 6 above)
Comment (Impala36) [inline src/main/java/Task.java:20]: Perhaps 'printing' could be presented as a present simple verb, like 'print'?

**Review by Impala36 — APPROVED** [review]

**Pull request closed without merging**